app/scripts/controllers/network/provider-api-tests/helpers.js

```diff
@@ -275,25 +289,35 @@ export async function withMockedInfuraCommunications(...args) {
 }
 
 /**
- * Builds a provider from the Infura middleware along with a block tracker, runs
- * the given function with those two things, and then ensures the block tracker
- * is stopped at the end.
+ * Builds a provider from the middleware (for the provider type) along with a
+ * block tracker, runs the given function with those two things, and then
+ * ensures the block tracker is stopped at the end.
  *
- * @param {WithInfuraClientArgs} args - Either an options bag + a function, or
- * just a function. The options bag, at the moment, may contain `network` (that
- * is, the Infura network; defaults to "mainnet"). The function is called with
- * an object that allows you to interact with the client via a couple of methods
- * on that object.
+ * @param {WithClientArgs} args - Either an options bag + a function, or
+ * just a function. The options bag, at the moment, may contain `network`
+ * (defaults to "mainnet"). The options bag may also include providerType
+ * (defaults to 'infura'). The function is called with an object that allows
+ * you to interact with the client via a couple of methods on that object.
  * @returns {Promise<any>} The return value of the given function.
  */
-export async function withInfuraClient(...args) {
+export async function withClient(...args) {
   const [options, fn] = args.length === 2 ? args : [{}, args[0]];
-  const { network = 'mainnet' } = options;
+  const { network = 'mainnet', type = 'infura' } = options;
 
-  const { networkMiddleware, blockTracker } = createInfuraClient({
-    network,
-    projectId: INFURA_PROJECT_ID,
-  });
+  let clientUnderTest;
+  if (type === 'infura') {
+    clientUnderTest = createInfuraClient({
+      network,
+      projectId: INFURA_PROJECT_ID,
+    });
+  } else {
+    clientUnderTest = createJsonRpcClient({
+      rpcUrl: LOCALHOST_RPC_URL,
+      chainId: BUILT_IN_NETWORKS[network].chainId || '0x1',
```

Comment:
I have another branch on controllers repo that moves these tests into typescript, and so I will need to adjust this code over there anyways.

Reply:
Okay. I'm not totally convinced we have to do this but perhaps there's something I'm not seeing. Maybe we change this over there in any case so those changes go together?

Reply:
Valid points. I thought it would be best to try to keep the scope of change as minimal as possible until we are using network controller from controllers repo.

Reply:
That said, I would like to unify the interfaces in the controllers repo PR that is soon to come...

Reply:
hmm... Ill give it a shot and see

Reply:
Leave a reaply